tests/test_bgp_af.rb

```diff
@@ -118,114 +157,151 @@ def test_default_information_originate
-    #
-    # Default is 'client-to-client' is configured
-    #
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    assert_nil(bgp_af.client_to_client,
+               'client_to_client should return nil on XR with non-default vrf')
+    assert_raises(Cisco::UnsupportedError) do
+      bgp_af.client_to_client = true
+    end
+  end
 
-    assert_match(pattern, af_string,
-                 "Error: 'client-to-client reflection' is not configured " \
-                   'and should be')
+  ##
+  ## client-to-client reflection (BT: trying a different style)
+  ##
+  def test_client_to_client
+    asn = '55'
+    vrf = 'red'
+    af = %w(ipv4 unicast)
 
-    assert(bgp_af.client_to_client,
-           "Error: 'client-to-client is not configured but should be")
-    #
-    # Unset and verify
-    #
+    # Setup XR?
+    if platform == :ios_xr
+      config_ios_xr_dependencies(asn)
+      client_to_client_vrf_xr_neg_test
+      vrf = 'default'
+    end
 
-    # Do a 'no client-to-client reflection'
-    bgp_af.client_to_client = false
-    pattern = /^ *no client-to-client reflection$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    # Create Address Family
+    bgp_af = RouterBgpAF.new(asn, vrf, af)
 
-    assert_match(pattern, af_string,
-                 "Error: 'no client-to-client' is not configured and should be")
+    # Check default value
+    initial = bgp_af.client_to_client
+    default = bgp_af.default_client_to_client
+    assert_equal(default, initial)
 
-    #
-    # Set and verify
-    #
+    # Toggle the state a few times
+    bgp_af.client_to_client = !initial
+    assert_equal(!initial, bgp_af.client_to_client)
 
-    # Do a 'client-to-client reflection'
-    bgp_af.client_to_client = true
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    bgp_af.client_to_client = initial
+    assert_equal(initial, bgp_af.client_to_client)
 
-    refute_match(pattern, af_string,
-                 "Error: 'no client-to-client' is configured and should not be")
+    bgp_af.destroy
   end
 
   ##
-  ## next_hop route-map
+  ## Negative tests with non-default vrf on ios_xr
   ##
-  def test_next_hop_route_map
+  def next_hop_route_policy_vrf_xr_neg_test
     asn = '55'
     vrf = 'red'
     af = %w(ipv4 unicast)
-
-    #
-    # Set and verify
-    #
     bgp_af = RouterBgpAF.new(asn, vrf, af)
-    bgp_af.next_hop_route_map = 'drop_all'
-    assert_match(bgp_af.next_hop_route_map, 'drop_all',
-                 'Error: nexthop route-map not set')
-    pattern = /^ *nexthop route-map drop_all$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    assert_nil(bgp_af.next_hop_route_policy,
+               'next_hop_route_policy should return ' \
+               'nil on XR with non-default vrf')
+    assert_raises(Cisco::UnsupportedError) do
+      bgp_af.next_hop_route_policy = '  drop_all  '
+    end
+  end
+
+  ##
+  ## next_hop route-map or route-policy
+  ##
+  def test_next_hop_route_map_or_policy
+    asn = '55'
```

Comment:
Wasn't sure how much of the code to muck with for this story.  It does seem very redundant.